### PR TITLE
Add adversarial_accuracy as a metric

### LINKED ISSUE #1775: A metric that just launch an attack and return the success rate (or the model accuracy).
Adding a simple alias to repeated code that launch an attack and returns the success rate into https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/metrics/metrics.py can be useful.

It can be as simple as the following.

```
def empirical_robustness(
    classifier: "CLASSIFIER_TYPE",
    x: np.ndarray,
    attack_name: str,
    attack_params: Optional[Dict[str, Any]] = None,
) -> float:
    crafter = get_crafter(classifier, attack_name, attack_params)
    crafter.set_params(**{"minimal": True})
    adv_x = crafter.generate(x)

    # Predict the labels for adversarial examples
    y = classifier.predict(x)
    y_pred = classifier.predict(adv_x)

    idxs = np.argmax(y_pred, axis=1) != np.argmax(y, axis=1)
    return np.sum(idxs) / len(y)
```

This can make it explicit to use an attack to "measure" the robustness for attack-based testing.

### PR DESCRIPTION
**Pull request by TS-Lee**
# Description

This PR adds `adversarial_accuracy` as discussed in #1775.

Fixes #1775

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] Test A
- [x] Test B

**Test Configuration**:
- OS: Ubuntu 18.04.6 LTS
- Python version: 3.8.12
- ART version or commit number: 50ade3b62836ec8588d0c88fbd0cad56675cc58d
- TensorFlow / Keras / PyTorch / MXNet version: Tensorflow==2.4.1, Keras==2.7.0, PyTorch==1.10.2

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by TS-Lee** [comment]
I think it is ready for review. Do you see some documentation to add this? Also, do you think this interface would be just enough for most cases?

File: art/metrics/metrics.py [hunk 1]
```diff
@@ -73,12 +78,53 @@ def get_crafter(classifier: "CLASSIFIER_TYPE", attack: str, params: Optional[Dic
...
+        crafter.set_params(**SUPPORTED_METHODS[attack]["params"])
     if params:
         crafter.set_params(**params)
 
     return crafter
 
 
+def adversarial_accuracy(
+    classifier: "CLASSIFIER_TYPE",
+    x: np.ndarray,
```
Comment (beat-buesser) [inline art/metrics/metrics.py:91]: Should we also provide the option to provide true labels `y`?

File: art/metrics/metrics.py [hunk 2]
```diff
@@ -73,12 +78,53 @@ def get_crafter(classifier: "CLASSIFIER_TYPE", attack: str, params: Optional[Dic
...
+) -> float:
+    """
+    Compute the adversarial accuracy of a classifier object over the sample `x` for a given adversarial crafting
+    method `attack`. `attack_name` can be specified to use a preset attack, or `attack_crafter` for wider choices
+    and customized parameters. Note that the score does not exclude wrong predictions by the classifier.
+
+    :param classifier: A trained model.
+    :param x: Input samples of shape that can be fed into `classifier`.
+    :param attack_name: A string specifying the attack to be used. Currently supported attacks are {`fgsm', `hsj`}
+                        (Fast Gradient Sign Method, Hop Skip Jump).
```
Comment (beat-buesser) [inline art/metrics/metrics.py:104]: I think we need to update the available attacks.

File: art/metrics/metrics.py [hunk 3]
```diff
@@ -32,17 +32,22 @@
...
 if TYPE_CHECKING:
-    from art.attacks.attack import EvasionAttack
     from art.utils import CLASSIFIER_TYPE, CLASSIFIER_LOSS_GRADIENTS_TYPE, CLASSIFIER_CLASS_LOSS_GRADIENTS_TYPE
 
 logger = logging.getLogger(__name__)
 
 SUPPORTED_METHODS: Dict[str, Dict[str, Any]] = {
+    "auto": {
+        "class": AutoAttack,
+        "params": {"eps_step": 0.1, "eps_max": 1.0},
```
Comment (beat-buesser) [inline art/metrics/metrics.py:49]: The default parameters most likely assume and input value range of [0, 1]. Should we raise a warning if we detect input values that are much larger than 1?

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @TS-Lee Thank you very much! I have added a few questions above and changed the target branch to `dev_1.12.0`. What do you think?

File: art/metrics/metrics.py (hunk 3 above)
Comment (TS-Lee) [reply art/metrics/metrics.py:49]: Added the check.

File: art/metrics/metrics.py (hunk 1 above)
Comment (TS-Lee) [reply art/metrics/metrics.py:91]: Added the option.

File: art/metrics/metrics.py (hunk 2 above)
Comment (TS-Lee) [reply art/metrics/metrics.py]: I changed the description to, basically refer to SUPPORTED_METHODS as it can evolve. I also modified the code to print available attack names when the input was wrong.

**Comment by TS-Lee** [comment]
@beat-buesser I see an error saying "certificate has expired". I'm not sure how to fix this. Do you have any idea/suggestions?

**Comment by beat-buesser** [comment]
@TS-Lee I have restarted the failing runs.

**Comment by TS-Lee** [comment]
Great. All checks have passed :)

**Review by TS-Lee — COMMENTED** [review]
Sure!

**Review by TS-Lee — COMMENTED** [review]
Is there anything I can do for this PR?

**Review by beat-buesser — APPROVED** [review]
Hi @TS-Lee Thank you very much! Looks good to me.
